java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -470,6 +471,42 @@ public static HostColumnVector fromStrings(String... values) {
     });
   }
 
+
+  /**
+   * Create a new vector from the given values.  This API supports inline nulls, but it is inefficient.
+   * Notice all input BigDecimals should share same scale.
+   */
+  public static HostColumnVector fromBigDecimals(BigDecimal... values) {
+    // 10000 is used as a default value of scale since no valid scale exceeds 19.
+    int scale = 10000;
+    DType.DTypeEnum typeId = DType.DTypeEnum.DECIMAL32;
+    for (BigDecimal v : values) {
+      if (v == null) {
+        continue;
+      }
+      if (scale == 10000) {
+        scale = v.scale();
+      }
+      // check scale consistence
+      assert v.scale() == scale : "Inconsistent decimal scale: " + scale + " vs " + v.scale();
+      // determine the underlying typeId according to maximum precision of inputs.
+      if (v.precision() > DType.DECIMAL64_MAX_PRECISION) {
```

Comment:
Are the max precision constants correct?  For decimal32 the max precision is 10, yet 10^10 is larger than 2^31. If we tried to store 10^10 into a 32-bit value it would overflow into the sign bit.  Similarly decimal64 precision max precision is 19 yet 10^19 is larger than 2^63.  It would be good to add some unit tests verifying the maximum/mininum extreme values representable by `BigDecimal` for what we think fits in `DECIMAL32` or `DECIMAL64` survive a round-trip through cudf and back to `BigDecimal`.

Reply:
You are right. Original max precision constants(10 for 32, 19 for 64) is not safe. Some extreme numbers will lead to overflow. So, I have shrinked the constants by 1 digit. For now, it is 10 for Decimal32, 18 for Decimal64.

Reply:
In terms of rounding, there exists no potenial rounding either from BigDecimal to cuDF or from cuDF to BigDecimal. We extract the exact unscaled value of BigDecimal, putting it to cuDF memory buffer. And we keep scale unchanged. Do you we need to support rounding extreme `BigDecimal`s to fit them into cuDF?

Reply:
> Do you we need to support rounding extreme BigDecimals to fit them into cuDF?

No, that implies we're losing precision.  Trying to place `BigDecimal` values that have more precision that we can represent should remain an error.  If the user wants the values to be rounded before being added to the column then that's something they can do directly.